### PR TITLE
feat(py): Expose DataCategory via C-ABI

### PR DESCRIPTION
**Pull request by jan-auer**
Exposes `DataCategory` via the C-ABI to Python for consumption in Sentry and Snuba. Along with this, also exposes the `SpanStatus`, still via the old dictionary. In a future change, this could be made an enum, too.

To expose the constants, they have been moved to `relay-cabi`. Unfortunately, a limitation in cbindgen does not allow to parse `relay-general` directly. On the upside, this creates a central place for constants that are exposed from Relay to other services in Sentry.

Along with this, the deprecated serialization of span status `"unknown_error"` is removed. By now, all downstream consumers have been updated.

### CODE REVIEW

File: py/sentry_relay/consts.py [hunk 1]
```diff
@@ -1,25 +1,98 @@
...
+            DataCategory.SECURITY,
+        )
+
+    @classmethod
+    def error_categories(cls):
+        """
+        Returns categories that count as traditional error tracking events.
+        """
+        return frozenset(
+            DataCategory.DEFAULT, DataCategory.ERROR, DataCategory.SECURITY
```
Comment (jan-auer) [inline py/sentry_relay/consts.py]: Note that this changes compared to Sentry, where `SECURITY` was not included:

https://github.com/getsentry/sentry/blob/9f08305e09866c8bd6d0c24f5b0aabdd7dd6c59c/src/sentry/constants.py#L532-L533

This is mainly used by quotas, which will now include security events as originally intended.

File: relay-common/src/constants.rs [hunk 2]
```diff
@@ -0,0 +1,290 @@
...
+            SpanStatus::DeadlineExceeded => write!(f, "deadline_exceeded"),
+            SpanStatus::Unauthenticated => write!(f, "unauthenticated"),
+            SpanStatus::PermissionDenied => write!(f, "permission_denied"),
+            SpanStatus::NotFound => write!(f, "not_found"),
+            SpanStatus::ResourceExhausted => write!(f, "resource_exhausted"),
+            SpanStatus::InvalidArgument => write!(f, "invalid_argument"),
+            SpanStatus::Unimplemented => write!(f, "unimplemented"),
+            SpanStatus::Unavailable => write!(f, "unavailable"),
+            SpanStatus::InternalError => write!(f, "internal_error"),
+            SpanStatus::Unknown => write!(f, "unknown"),
```
Comment (jan-auer) [inline relay-common/src/constants.rs:281]: This is now changed from `"unknown_error"`, along with all other uses.

File: relay-server/src/utils/unreal.rs [hunk 3]
```diff
@@ -178,7 +178,10 @@ fn merge_unreal_context(event: &mut Event, context: Unreal4Context) {
     // Clear this property. It's a duplicate of misc_primary_gpu_brand and only exists because
     // somebody made a typo once while mapping the GPU data from unrealcontext in symbolic, and we
     // tried to fix this in a backwards-compatible way by retaining both properties.
-    runtime_props.misc_primary_cpu_brand = None;
+    #[allow(deprecated)]
+    {
+        runtime_props.misc_primary_cpu_brand = None;
```
Comment (jan-auer) [inline relay-server/src/utils/unreal.rs:183]: This is fixes an existing warning, unrelated to this patch.

File: py/sentry_relay/consts.py [hunk 4]
```diff
@@ -1,25 +1,92 @@
...
+
+
+def _make_span_statuses():
+    prefix = "RELAY_SPAN_STATUS_"
+
+    for attr in dir(lib):
+        if not attr.startswith(prefix):
+            continue
+
+        status_name = attr[len(prefix) :].lower()
```
Comment (jan-auer) [inline py/sentry_relay/consts.py:77]: I would say this is actually safe - per convention the constants must always match the default serialization. That said, if we ever diverge from this in the future (which we should not!) we can call into an API for this.

File: py/sentry_relay/consts.py [hunk 5]
```diff
@@ -1,25 +1,93 @@
+from enum import IntEnum
+
+from sentry_relay._compat import int_enum_base
```
Comment (untitaker) [inline py/sentry_relay/consts.py]: Can we just have a regular class? I think it's also problematic that we change behavior drastically between py2 vs 3. What's the purpose of subclassing from int?

Comment (untitaker) [reply py/sentry_relay/consts.py]: Also I don't understand why we use `int_enum_base` for one enum and enum34 for the other.

File: py/sentry_relay/consts.py (hunk 4 above)
Comment (untitaker) [reply py/sentry_relay/consts.py:77]: Yes this seems safe.

**Review by untitaker — APPROVED** [review]
I think the changes to the default serialization of SpanStatus should be deployed with extreme care.

File: py/sentry_relay/consts.py (hunk 5 above)
Comment (jan-auer) [reply py/sentry_relay/consts.py]: `DataCategory` was an `IntEnum` in Sentry already, and we rely on the behavior that's also reflected in tests, for example:

```py
1 == DataCategory.ERROR
```

We *also* need the class methods for backwards compatibility (and generally as part of the usage of `DataCategory`. Now, the problem is that there seems to be incompatibility between Python 2 and 3. If I make the base class `int` in Python 3, it cannot create the enum. If I make the base class `object` (or anything else) in Python 2, it cannot create the enum.

Comment (jan-auer) [reply py/sentry_relay/consts.py]: Maybe this is not actually a difference between Python 2 and 3, but rather between builtin `enum` and `enum34`.

Comment (untitaker) [reply py/sentry_relay/consts.py]: If you only care about the equality stuff (and not the actual type relations) I'd recommend implementing the `__eq__` method.

Comment (untitaker) [reply py/sentry_relay/consts.py]: Actually ignore, `IntEnum` already subclasses from `int`.

Comment (untitaker) [reply py/sentry_relay/consts.py]: https://bitbucket.org/stoneleaf/enum34/src/c208549a93b71b948ff7bbdfd29dce8f85527916/enum/__init__.py#lines-819

**Review by untitaker — APPROVED** [review]

File: py/sentry_relay/consts.py (hunk 5 above)
Comment (jan-auer) [reply py/sentry_relay/consts.py]: Thanks, updated. Programmatically adding to the class avoids the mixin class and is overall much cleaner.

**Pull request merged**